baldur/src/intTest/java/org/opencadc/baldur/GetPermissionsTest.java

```diff
@@ -229,17 +230,10 @@ public Object run() throws Exception {
                     URL getPermissionsURL = new URL(certURL.toString() + "?OP=read&uri=" + notFoundArtifactURL);
                     HttpDownload get = new HttpDownload(getPermissionsURL, out);
                     get.run();
-                    Assert.assertNull(get.getThrowable());
-                    Assert.assertEquals(200, get.getResponseCode());
-                    
-                    GrantReader grantReader = new GrantReader();
-                    Grant grant = grantReader.read(out.toString());
-                    Assert.assertTrue(grant instanceof ReadGrant);
-                    ReadGrant readGrant = (ReadGrant) grant;
-                    Assert.assertEquals(URI.create(notFoundArtifactURL), readGrant.getArtifactURI());
-                    Assert.assertFalse(readGrant.isAnonymousAccess());
-                    Assert.assertTrue(readGrant.getGroups().isEmpty());
-                    
+                    Assert.assertEquals(404, get.getResponseCode());
```

Comment:
Please change to use HttpGet anywhere that is not downloading a file to local disk.

Checking the type of exception is only verifying the client behaviour. The response code 404 is sufficient for correct service behaviour.

Reply:
Updated to use HttpGet.

Reply:
Updated to properly use HttpGet.